compiler/front/options.nim

```diff
@@ -1493,19 +1480,23 @@ proc findProjectNimFile*(conf: ConfigRef; pkg: string): string =
   return ""
 
 proc canonicalImportAux*(conf: ConfigRef, file: AbsoluteFile): string =
-  ## Shows the canonical module import, e.g.: system, std/tables,
-  ## fusion/pointers, system/assertions, std/private/asciitables
-  var ret = getRelativePathFromConfigPath(conf, file, isTitle = true)
-  let dir = getNimbleFile(conf, $file).parentDir.AbsoluteDir
-  if not dir.isEmpty:
-    let relPath = relativeTo(file, dir)
-    if not relPath.isEmpty and (ret.isEmpty or relPath.string.len < ret.string.len):
-      ret = relPath
-  if ret.isEmpty:
-    ret = relativeTo(file, conf.projectPath)
-  result = ret.string
+  ## canonical module import filename, e.g.: system.nim, std/tables.nim,
+  ## system/assertions.nim, etc
+  let
+    desc = getPkgDesc(conf, file.string)
+    (_, moduleName, ext) = file.splitFile
+  if desc.pkgKnown:
```

Comment:
```suggestion
  if desc.pkgKnown and
     AbsoluteFile(getNimbleFile(conf, conf.projectFull.string)) != desc.pkgFile:
```
This fixes the test failures, and is also a sensible rule in general, I think. Essentially, if module A belongs to the same package as the project file, the package qualifier is not included in the canonical path.

Side-note: a call to `toAbsolute` might be needed for the result of `getNimbleFile`, both here and in `getPkgDesc`.

Reply:
Hahaha, this is the same conclusion I came to and asked about in the other comment, maybe drop the package root if it's shared. 😆